File: Data Structures & Algorithms/insertionSort/submission-1.java
// Definition for a pair
// class Pair {
//     int key;
//     String value;
//
//     Pair(int key, String value) {
//         this.key = key;
//         this.value = value;
//     }
// }
public class Solution {
    public List<List<Pair>> insertionSort(List<Pair> pairs) {
        List<List<Pair>> result = new ArrayList<List<Pair>>();
        if (pairs.size() == 0){
            return result;
        }
        
        result.add(new ArrayList<Pair>(pairs));

        for (int idx = 1; idx < pairs.size();idx++){
            int iterator = idx -1;
            while(iterator >=0 && pairs.get(iterator+1).key < pairs.get(iterator).key){
                Pair temp = pairs.get(iterator+1);
                pairs.set(iterator+1, pairs.get(iterator));
                pairs.set(iterator,temp);
                iterator--;
            }

            result.add(new ArrayList<Pair>(pairs));
        }

        return result;
    }
}

File: Data Structures & Algorithms/add-two-numbers/submission-1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        
        rem = 0
        result = ListNode()
        temp = result

        while l1 and l2:
            val = l1.val + l2.val+rem
            if val >=10:
                rem = 1
                val= val %10
            else:
                rem=0
            result.next = ListNode(val)
            result = result.next
            l1 =l1.next
            l2 =l2.next
        
        while l1:
            val = l1.val +rem

            if val >=10:
                rem = 1
                val= val %10
            else:
                rem=0
            result.next = ListNode(val)
            result= result.next
            l1=l1.next
        
        while l2:
            val = l2.val+rem

            if val >=10:
                rem = 1
                val = val %10
            else:
                rem=0

            result.next = ListNode(val)
            result= result.next
            l2 = l2.next

        if rem == 1:
            result.next = ListNode(rem)

        return temp.next


            
